### PR TITLE
 Add flag to avoid fail upon nothing to commit in Commit and push action

### LINKED ISSUE #87: Add flag to avoid fail upon `nothing to commit` in Commit and push action
Change must not be breaking, default behavior must remain as before.

Proposed is a flag `pass-empty-commit`.
The default value would be `false`.

### PR DESCRIPTION
**Pull request by sujuka99**
closes #87

### CODE REVIEW

**Comment by sujuka99** [comment]
Left:
- [ ] Test
- [ ] Document

File: actions/commit-and-push/action.yaml [hunk 1]
```diff
@@ -39,10 +43,16 @@ runs:
...
 
-    - name: Commit files
+    - name: Commit files, exit 1 if nothing to commit
+      if: "! inputs.pass-empty-commit"
       run: git commit -m "${{ inputs.commit-message }}" -a
       shell: bash
 
+    - name: Commit files, exit 0 if nothing to commit and no uncommitted changes
+      if: "inputs.pass-empty-commit"
+      run: '( [ -z "$(git status --porcelain=v1 2>/dev/null)"] && echo "No uncommitted changes." ) || git commit -m "${{ inputs.commit-message }}" -a'
```
Comment (yannick-roeder) [inline actions/commit-and-push/action.yaml:53]: Bash doesn't normally like `[` and `]` without spaces. Without it, it throws a warning locally for me:
```suggestion
      run: '( [ -z "$(git status --porcelain=v1 2>/dev/null)" ] && echo "No uncommitted changes." ) || git commit -m "${{ inputs.commit-message }}" -a'
```

File: actions/commit-and-push/action.yaml [hunk 2]
```diff
@@ -22,6 +22,10 @@ inputs:
     description: "Whether to add untracked files to commit."
     required: false
     default: "false"
+  pass-empty-commit:
+    description: "Whether to exit with code 0 when nothing to commit."
+    required: false
+    default: "false"
```
Comment (yannick-roeder) [inline actions/commit-and-push/action.yaml:28]: A boolean would be better here and GitHub actions supports it. Please shortly test again after changing:
```suggestion
    default: false
```

**Review by yannick-roeder — CHANGES_REQUESTED** [review]

File: actions/commit-and-push/action.yaml (hunk 2 above)
Comment (sujuka99) [reply actions/commit-and-push/action.yaml]: WITH the quotes:

[pass-empty-commit=true, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594742711/jobs/8114152450)
[pass-empty-commit=true, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594745065/jobs/8114158026)
[pass-empty-commit=false, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594727019/jobs/8114115428)
[pass-empty-commit=false, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594731254/jobs/8114125237)


WITHOUT quotes:

[pass-empty-commit=true, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594801621)
[pass-empty-commit=true, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594808915)
[pass-empty-commit=false, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594823872)
[pass-empty-commit=false, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594831337)

**Comment by sujuka99** [comment]
WITH quotes for `default`:

[pass-empty-commit=true, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594742711/jobs/8114152450)
[pass-empty-commit=true, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594745065/jobs/8114158026)
[pass-empty-commit=false, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594727019/jobs/8114115428)
[pass-empty-commit=false, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594731254/jobs/8114125237)


WITHOUT quotes for `default`:

[pass-empty-commit=true, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594801621/jobs/8114288219)
[pass-empty-commit=true, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594808915/jobs/8114302268)
[pass-empty-commit=false, Non-empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594823872/jobs/8114337009)
[pass-empty-commit=false, Empty commit](https://github.com/bakdata/github-playground-2/actions/runs/4594831337/jobs/8114352501)

**Pull request merged**